docs/overview.rst

```diff
@@ -451,3 +451,71 @@ infrastructure, Linux and SSH. A good place to start is Heroku (https://www.hero
 We reference Heroku in our tutorials, but Agile Toolkit can work with any cloud
 hosting that runs PHP apps.
 
+
+Online-Demo showing many functions
+==================================
+
+There's a demo available of atk4/ui which shows many of the modules and functions available
+in atk4.
+You can watch & use the source code of each example to find best practice examples and to see
+how to use atk4 in certain application cases.
+
+You can find the demo here:
+https://agile-ui.readthedocs.io/en/latest/index.html
+
+
+Demo contained in the atk4 repository
+=====================================
+When you download and install atk4 you will find a subdirectory called "demos" in the atk4
+repository which also could be locally executed.
+This folder is basically intended to run under the UI package only.
+That means the demo may not work properly, if you copy the "demos" directory out of this location
+(to run it from a different location) or if the "vendors" directory isn't accessible through the
+webserver.
+But if you change some files (e.g. the database setup file) and you do a composer update, your
+changes would be overwritten.
+
+.. important:: We strictly advise you to clone the whole atk4/ui repo into a separate directory.
+
+Here's how that works:
+
+Pre-requirements
+----------------
+
+ - You have setup a (local) webserver or Docker instance (accessible e.g. by "https://localhost")
+ - The forked atk4/ui directory is within the publicly accesible root directory of the webserver (that means it could for example be opened through: "https://localhost/atk4/ui/demos")
+
+Setup the demo
+--------------
+
+To run the demo, we recommend to...
+ - Create a directory called "atk4" and create a separate folder for each repo (ui, data, etc.), in this case "ui"
+ - Fork the original repo into this directory
+ - Copy the file "db.default.php" from the "atk4/ui/demos" subdirectory
+ - Rename the copied file to "db.php"
+ - Open the renamed file and edit your database details accordingly to fit to your database
+ - Setup the database using the file "dump.sql" from the subdirectory "atk4/ui/demos/_demo-data" of the "demos" directory. You can also setup an SQLITE database using a provided script (see below)
+ - Open the demos from your browser (e.g. with https://localhost/vendors/atk4/ui/demos or if you choose to clone it, with e.g. https://localhost/atk4/ui/demos)
+
+Setup SQLITE database
+---------------------
+The demo also includes a script that let's you setup a SQLITE database. You will find this script in the subdirectory
+"atk4/ui/demos/_demo-data/".
+Call it as follows (you may have to adapt the paths to fit your webserver structure)::
+
+    php atk4/ui/demos/_demo-data/create-sqlite-db.php;  // when using the CLI approach
```

Comment:
we should allow to create from CLI only

Reply:
it might be more confusing to offer alternative implementation, but if your opinion as someone new to ATK that both instruction should be provided -that's OK.. 

Just consider file permission implications. 12-factor app requires applications to be runnable in read-only containers.

Same can be solved with docker-compose and mysql container, btw.